File: src/leetcode/bit/java/ReverseBits.java
package leetcode.bit.java;

/*
    time: O(1)
    space: O(1)
 */
public class ReverseBits {
    public int reverseBits(int n) {
        int res = 0;
        for (int i = 0; i < 32; i++) {
            res <<= 1;
            if ((n & 1) == 1) {
                res++;
            }
            n >>= 1;
        }
        return res;
    }
}

File: src/leetcode/bit/kotlin/ReverseBits.kt
package leetcode.bit.kotlin

fun reverseBits(n:Int):Int {
    var res = 0
    var num = n
    for (i in 1..32) {
        res = res shl 1
        if ((num and 1) == 1) res++
        num = num shr 1
    }
    return res;
}